### PR TITLE
lux: init at 0.15.0

### PR DESCRIPTION
**Pull request by ghost**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/video/lux/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  buildPhase = ''
+    buildFlagsArray=(-v -p $NIX_BUILD_CORES -ldflags="-s -w")
+    runHook preBuild
+    go build "''${buildFlagsArray[@]}" -o lux
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    install -Dm755 lux -t $out/bin
+  '';
```
Comment (figsoda) [inline pkgs/tools/video/lux/default.nix:26]:
```suggestion
  ldflags = [ "-s" "-w" ];
```

**Review by figsoda — CHANGES_REQUESTED** [review]

File: pkgs/tools/video/lux/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{ lib, buildGoModule, fetchFromGitHub, ffmpeg }:
```
Comment (figsoda) [inline pkgs/tools/video/lux/default.nix:1]:
```suggestion
{ lib, buildGoModule, fetchFromGitHub, makeWrapper, ffmpeg }:
```

File: pkgs/tools/video/lux/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  version = "0.15.0";
+  src = fetchFromGitHub {
+    owner = "iawia002";
+    repo = "lux";
+    rev = "v${version}";
+    sha256 = "sha256-fZR+Q0duITZq3Ynr2WTZAhDnmEkXrT2gXUlpuN0+aFo=";
+  };
+
+  buildInputs = [ ffmpeg ];
+
```
Comment (figsoda) [inline pkgs/tools/video/lux/default.nix:14]:
```suggestion
```

File: pkgs/tools/video/lux/default.nix [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  vendorSha256 = "sha256-SHUtyfGRGriEaESo6th7gGQn6V4REdk3XT0ZlGwky7E=";
+
+  ldflags = [ "-s" "-w" ];
+
+  doCheck = false;
+
```
Comment (figsoda) [inline pkgs/tools/video/lux/default.nix:20]:
```suggestion

  postInstall = ''
    wrapProgram $out/bin/lux \
      --prefix PATH : ${lib.makeBinPath [ ffmpeg ]}
  '';

```

File: pkgs/tools/video/lux/default.nix [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    owner = "iawia002";
+    repo = "lux";
+    rev = "v${version}";
+    sha256 = "sha256-fZR+Q0duITZq3Ynr2WTZAhDnmEkXrT2gXUlpuN0+aFo=";
+  };
+
+  buildInputs = [ ffmpeg ];
+
+  vendorSha256 = "sha256-SHUtyfGRGriEaESo6th7gGQn6V4REdk3XT0ZlGwky7E=";
+
```
Comment (figsoda) [inline pkgs/tools/video/lux/default.nix:16]:
```suggestion

  nativeBuildInputs = [ makeWrapper ];

```

**Review by figsoda — CHANGES_REQUESTED** [review]
Sorry that I missed this in my first review. `ffmpeg` being in `buildInputs` doesn't work as a runtime dependencies and needs to be added to `PATH` with a wrapper

**Comment by candyc1oud** [comment]
I fixed it.

**Review by figsoda — APPROVED** [review]

**Pull request merged**